lib/elixir/lib/map_set.ex

```diff
@@ -196,19 +198,17 @@ defmodule MapSet do
     {map1, map2} = order_by_size(map1, map2)
 
     map1
-    |> Map.keys()
+    |> :maps.iterator()
+    |> :maps.next()
     |> none_in?(map2)
   end
 
-  defp none_in?([], _) do
-    true
-  end
+  @compile {:inline, [none_in?: 2]}
```

Comment:
We probably shouldn't inline recursive functions, the in-liner is sometimes pretty weird with them.

Reply:
I ran the benchmarks again with the inlining removed, it made a negligible difference. I've removed the inline declaration for `none_in?/2` and `all_in?/2`.